### DISCUSSION TITLE
How to change colour of cursor for Insert and Normal Modes

### DISCUSSION THREAD

**Discussion by andynameistaken** [vscode-neovim/vscode-neovim#2237] [Q&A]
Hi, how to setup properly cursor colours for neovim modes?

**Comment by justinmk** [comment]
'guicursor' isn't fully supported yet: https://github.com/vscode-neovim/vscode-neovim/issues/1876

however, the Nvim mode is set as a vscode `setContext` key: https://github.com/vscode-neovim/vscode-neovim/blob/6f2fa02733330d44c624e87c251f68ad80631c74/src/mode_manager.ts#L112 

, so if vscode supports a `when` clause for configuring its cursor, you could use the `neovim.mode` setContext key to control the `when` clause.

Please share your solution here if you find one.

**Comment by wrath-codes** [comment]
i wrote an extension for that,, if that help:
[Nvim UI+](https://marketplace.visualstudio.com/items?itemName=wrathcodes.nvim-ui-plus)
